lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -360,24 +365,57 @@ private Optional<JourneyResponse> checkForProfileMatch(
     }
 
     private JourneyResponse buildF2FNoMatchResponse(
-            boolean areGpg45VcsCorrelated, AuditEventUser auditEventUser) throws SqsException {
+            boolean areGpg45VcsCorrelated,
+            AuditEventUser auditEventUser,
+            ContraIndicators contraIndicators)
+            throws SqsException, MitigationRouteConfigNotFoundException, ConfigException,
+                    UnsupportedMitigationRouteException {
         LOGGER.info(LogHelper.buildLogMessage("F2F return - failed to match a profile."));
         sendAuditEvent(
                 !areGpg45VcsCorrelated
                         ? AuditEventTypes.IPV_F2F_CORRELATION_FAIL
                         : AuditEventTypes.IPV_F2F_PROFILE_NOT_MET_FAIL,
                 auditEventUser);
+        var mitigatedCI = ciMitUtilityService.hasMitigatedContraIndicator(contraIndicators);
+        if (mitigatedCI.isPresent()) {
+            var mitigationRoute = ciMitUtilityService.getMitigatedCiJourneyStep(mitigatedCI.get());
+            if (mitigationRoute.isPresent()) {
+                JourneyResponse journeyResponse = mitigationRoute.get();
+                if (!JOURNEY_ENHANCED_VERIFICATION_PATH.equals(journeyResponse.getJourney())) {
+                    throw new UnsupportedMitigationRouteException(
+                            String.format(
+                                    "Unsupported mitigation route: %s",
+                                    journeyResponse.getJourney()));
+                }
+                return JOURNEY_ENHANCED_VERIFICATION_F2F_FAIL;
+            }
+        }
         return JOURNEY_F2F_FAIL;
     }
 
     private JourneyResponse buildNoMatchResponse(
-            List<VerifiableCredential> verifiableCredentials, AuditEventUser auditEventUser)
-            throws SqsException {
+            List<VerifiableCredential> verifiableCredentials,
+            AuditEventUser auditEventUser,
+            ContraIndicators contraIndicators)
+            throws SqsException, MitigationRouteConfigNotFoundException, ConfigException,
+                    UnsupportedMitigationRouteException {
+
+        var mitigatedCI = ciMitUtilityService.hasMitigatedContraIndicator(contraIndicators);
+        if (mitigatedCI.isPresent()) {
+            var mitigationRoute = ciMitUtilityService.getMitigatedCiJourneyStep(mitigatedCI.get());
+            if (mitigationRoute.isEmpty()) {
+                throw new UnsupportedMitigationRouteException(
+                        String.format(
+                                "Empty mitigation route for mitigated CI: %s", mitigatedCI.get()));
+            }
+            return mitigationRoute.get();
```

Comment:
Because you're using optionals here, you could avoid the if statement by doing something like this:
```
return ciMitUtilityService
                    .getMitigatedCiJourneyStep(mitigatedCI.get())
                    .orElseThrow(
                            () ->
                                    new UnsupportedMitigationRouteException(
                                            String.format(
                                                    "Empty mitigation route for mitigated CI: %s",
                                                    mitigatedCI.get())));
```